src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -406,19 +406,38 @@ public ResultSet getTableTypes() throws SQLException {
 
     // Helper for getting a stream of all database names.
     private Stream<String> getDatabaseNames() {
-        return this.conn.mongoClient.listDatabaseNames().into(new ArrayList<>()).stream();
+        return this.conn
+                .mongoClient
+                .listDatabaseNames()
+                .into(new ArrayList<>())
+                .stream()
+                .filter(dbName -> !dbName.isEmpty());
+    }
+
+    // Helper for getting a list of collection names from the db
+    // Using runCommand instead of listCollections as listCollections does not support authorizedCollections option
+    private ArrayList<Document> getCollectionsFromRunCommand(MongoDatabase db) {
+        Document listCollectionsResult =
+                db.runCommand(
+                        new Document("listCollections", 1)
```

Comment:
Since we are now using a listCollections command, can we create a JIRA to add server side filtering using the `filter` field?
For both the JDBC and ODBC driver, please.

Reply:
Sure, created JIRAs for JDBC and ODBC drivers.